core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowAndLongStateSerializer.java

```diff
@@ -54,20 +60,31 @@ public void deserialize(Block block, int index, LongDecimalWithOverflowAndLongSt
     {
         if (!block.isNull(index)) {
             Slice slice = VARBINARY.getSlice(block, index);
-            if (slice.length() != SERIALIZED_SIZE) {
-                throw new IllegalStateException("Unexpected serialized state size: " + slice.length());
+            if (slice.length() == Long.BYTES) {
+                state.setLong(1);
+                state.setOverflow(0);
+                state.setNotNull();
+                long[] decimal = state.getDecimalArray();
+                int offset = state.getDecimalArrayOffset();
+                decimal[offset] = 0;
+                decimal[offset + 1] = slice.getLong(0);
             }
+            else {
```

Comment:
similar comment as in https://github.com/trinodb/trino/pull/13573/files#r942932796 but reverse